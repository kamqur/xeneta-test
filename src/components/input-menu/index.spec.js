import React from "react";
import { shallow } from "enzyme";

import InputMenu from "./index";

const InputMenuClass = "input-menu";

describe("Testing Input Menu Componenet", () => {
  it("snapshort testing", () => {
    const component = shallow(
      <InputMenu
        options={[{ code: "test", name: "test" }]}
        setValue={() => {}}
        label="test"
        currentValue=""
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("Add space", () => {
    const component = shallow(
      <InputMenu
        options={[{ code: "test", name: "test" }]}
        setValue={() => {}}
        label="test"
        currentValue=""
        space
      />
    );
    const spaceEl = component.find(`.${InputMenuClass}-space`);
    expect(spaceEl).toHaveLength(1);
  });

  it("No space is added", () => {
    const component = shallow(
      <InputMenu
        options={[{ code: "test", name: "test" }]}
        setValue={() => {}}
        label="test"
        currentValue=""
      />
    );
    const spaceEl = component.find(`.${InputMenuClass}-space`);
    expect(spaceEl).toHaveLength(0);
  });

  it("has select component", () => {
    const component = shallow(
      <InputMenu
        options={[{ code: "test", name: "test" }]}
        setValue={() => {}}
        label="test"
        currentValue=""
      />
    );
    const selectMenu = component.find(`.${InputMenuClass}-select-input`);
    expect(selectMenu).toHaveLength(1);
  });

  it("has no select component", () => {
    const component = shallow(
      <InputMenu setValue={() => {}} label="test" currentValue="" />
    );
    const selectMenu = component.find(`.${InputMenuClass}-select-input`);
    expect(selectMenu).toHaveLength(0);
  });

  it("setValue is called", () => {
    const onChange = jest.fn();

    const component = shallow(
      <InputMenu
        options={[{ code: "test", name: "test" }]}
        setValue={onChange}
        label="test"
        currentValue=""
      />
    );

    const selectMenu = component
      .find(`.${InputMenuClass}-select-input`)
      .simulate("change", { target: { code: "test", name: "test" } });
    expect(onChange).toHaveBeenCalled();
  });
});
