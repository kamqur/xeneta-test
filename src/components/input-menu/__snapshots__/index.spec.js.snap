// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input Menu Componenet snapshort testing 1`] = `
<div
  className="input-menu"
>
  <div
    className="input-menu-label"
  >
     
    test
     
  </div>
  <StateManager
    className="input-menu-select-input"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "test",
          "value": Object {
            "code": "test",
            "name": "test",
          },
        },
      ]
    }
  />
</div>
`;
